codalab/model/bundle_model.py

```diff
@@ -1334,7 +1336,37 @@ def add_worksheet_item(self, worksheet_uuid, item):
             'sort_key': None,
         }
         with self.engine.begin() as connection:
-            connection.execute(cl_worksheet_item.insert().values(item_value))
+            if after_sort_key is not None:
+                after_sort_key = int(after_sort_key)
+                # Find all the items that originally come after this after_sort_key.
+                clause = and_(
+                    cl_worksheet_item.c.worksheet_uuid == worksheet_uuid,
+                    cl_worksheet_item.c.sort_key > after_sort_key,
+                )
+                query = select(['*']).where(clause)
+                # Get results in list format.
+                after_items = [item for item in connection.execute(query)]
+                # check if there are gaps between the smallest sort_key among the
+                # after_items and our after_sort_key
+                if len(after_items) > 0 and min(item_sort_key(item) for item in after_items) - after_sort_key == 1:
+                    # There is no gap, space out the sort_keys to make some room
+                    connection.execute(cl_worksheet_item.delete().where(clause))
```

Comment:
I would move this into a separate function called `space_out_sort_keys` that is synchronized with `update_worksheet_items`...basically all the logic for setting sort_keys should be consolidated.  I think we need to be a bit careful we don't break any invariants.